src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -84,18 +84,25 @@ private byte[] EncryptOrDecrypt(byte[] data, RSAEncryptionPadding padding, Encry
         //
         // Now that the padding mode and information have been marshaled to their native counterparts, perform the encryption or decryption.
         //
-        private static unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, byte[] input, AsymmetricPaddingMode paddingMode, void* paddingInfo, EncryptOrDecryptAction encryptOrDecrypt)
+        private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, byte[] input, AsymmetricPaddingMode paddingMode, void* paddingInfo, EncryptOrDecryptAction encryptOrDecrypt)
         {
-            int numBytesNeeded;
-            ErrorCode errorCode = encryptOrDecrypt(key, input, input.Length, paddingInfo, null, 0, out numBytesNeeded, paddingMode);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            int estimatedSize = KeySize / 8;
```

Comment:
I obtained this my experimenting with a bunch of key sizes and padding algorithms. This was always on the nose for Encrypts and Signs and just a little above the needed size (due to padding) for Decrypts.